task8/ProfitApp/src/main/java/com/bsu/fpmi/ProfitApp/AdCollection.java

```diff
@@ -0,0 +1,180 @@
+package com.bsu.fpmi.ProfitApp;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AdCollection {
+
+    List<Ad> ads;
+
+    AdCollection() {
+        ads = new ArrayList<>();
+    }
+
+    public Ad get(String id) {
+        return ads.stream().filter(a -> a.getId().equals(id)).findAny().orElse(null);
+    }
+
+    public boolean remove(String id) {
+        return ads.removeIf(a -> a.getId().equals(id));
+    }
+
+    public List<Ad> getPage(int skip, int top, FilterConfig fc, String sortType) {
+        List<Ad> toReturn = ads;
+        SimpleDateFormat df = new SimpleDateFormat("dd.MM.yyyy");
+        if (sortType.equalsIgnoreCase("Rating")) {
+            toReturn = toReturn.stream().sorted((a, b) -> Double.compare(b.getRating(), a.getRating())).collect(Collectors.toList());
+        } else {
+            toReturn = toReturn.stream().sorted((a, b) ->
+                    {
+                        try {
+                            return df.parse(b.getCreatedAt()).compareTo(df.parse(a.getCreatedAt()));
+                        } catch (ParseException e) {
+                            e.printStackTrace();
+                        }
+                        return 0;
+                    }
+            ).collect(Collectors.toList());
+        }
+        if (fc.getFrom() != null) {
+            toReturn = toReturn.stream().filter(a -> {
+                try {
+                    return df.parse(a.getCreatedAt()).after(df.parse(fc.getFrom()));
+                } catch (ParseException e) {
+                }
+                return false;
+            }).collect(Collectors.toList());
+        }
+
+        if (fc.getTo() != null) {
+            toReturn = toReturn.stream().filter(a -> {
+                try {
+                    return df.parse(a.getCreatedAt()).before(df.parse(fc.getTo()));
+                } catch (ParseException e) {
+                }
+                return false;
+            }).collect(Collectors.toList());
+        }
+        if (fc.getVendor() != null) {
+            toReturn = toReturn.stream().filter(a -> a.getVendor().equals(fc.getVendor())).collect(Collectors.toList());
+        }
+        if (skip >= toReturn.size()) {
+            toReturn = null;
+        } else if (skip + top > toReturn.size()) {
+            top = toReturn.size() - skip;
+        }
+        if (toReturn != null) {
```

Comment:
`toReturn` cannot be null. A collector always returns at least an empty collection